backend/src/main/java/com/htecgroup/skynest/service/impl/FolderServiceImpl.java

```diff
@@ -136,6 +140,38 @@ public List<FolderResponse> getAllFoldersWithParent(UUID parentFolderId) {
     return asFolderResponseList(allFolders);
   }
 
+  @Override
+  public void moveFolderToBucket(MoveFolderToBucketRequest moveFolderToBucketRequest, UUID uuid) {
+    FolderEntity folderEntity =
+        folderRepository.findById(uuid).orElseThrow(FolderNotFoundException::new);
+    BucketEntity bucketEntity =
+        bucketRepository
+            .findById(moveFolderToBucketRequest.getDestinationBucketId())
+            .orElseThrow(BucketNotFoundException::new);
+    if (folderEntity.getBucket().getId() == bucketEntity.getId()) {
+      throw new FolderAlreadyInsideBucketException();
+    }
+    folderEntity.setBucket(bucketEntity);
+    folderRepository.save(folderEntity);
+    actionService.recordAction(Collections.singleton(folderEntity), ActionType.MOVE);
```

Comment:
Wasn't there an annotation that does this action logic?

Reply:
Yes, but it's on different PR, still not on dev.